src/armory/SetWagerPopup.js

```diff
@@ -3,93 +3,184 @@
 import * as React from 'react';
 import Popup from 'reactjs-popup';
 import Tank from '../tanks/Tank.js';
-import { setWager } from '../globalComponents/apiCalls/userAPIIntegration.js';
+import { setWager, setWager3v3 } from '../globalComponents/apiCalls/userAPIIntegration.js';
 import getUserAPICall from '../globalComponents/apiCalls/getUserAPICall';
-import { setFavoriteTankId, getFavoriteTank, removeFavoriteTankId } from '../globalComponents/apiCalls/tankAPIIntegration.js';
+import { 
+	setFavoriteTankId, 
+	getFavoriteTank,
+	removeFavoriteTankId, 
+	removeFavoriteTankTeamIds,
+	getFavoriteTankTeam, 
+	setFavoriteTankTeamIds,
+	getAllUsersTanks
+} from '../globalComponents/apiCalls/tankAPIIntegration.js';
 import { ToastContainer , toast } from 'react-toastify';
+import type { BattleType } from '../globalComponents/typesAndClasses/BattleType.js';
+import { getEmptyCasusCode, getTank } from '../tanks/TankLoader.js';
+import BackendTank from '../tanks/BackendTank.js';
 
 type Props = {|
-	wagerTank: Tank,
-	onWagerUpdate: () => void;
+	onWagerUpdate: () => void
 |};
 
 type State = {|
+	allTanks: Array<Tank>,
 	userCurrency: number,
 	userWager: number,
-	currentWager: number,
-	currentWagerTank: ?Tank,
+	userWager3v3: number,
+	newWager: number,
+	userWager3v3Tanks: Array<?Tank>,
+	userWager1v1Tank: ?Tank,
+	newWager3v3Tanks: Array<?Tank>,
+	newWager1v1Tank: Tank,
 	setWagerOpen: boolean,
 	removeWagerOpen: boolean,
+	battleType: BattleType
 |};
 
 class SetWagerPopup extends React.Component<Props, State> {
 	constructor(props: Props) {
 		super(props);
 
+		// Create a blank tank as a placeholder until tanks are pulled.
+		const blankTank: BackendTank = new BackendTank(
+			'',
+			['empty', 'empty', 'empty', 'empty', 'empty', 'empty', 'empty', 'empty', 'empty', 'empty', 'empty',],
+			getEmptyCasusCode(),
+			false,
+			'',
+			''
+		);
+
 		this.state = {
+			allTanks: [],
 			userCurrency: 0,
 			userWager: 0,
-			currentWager: 0,
-			currentWagerTank: null,
+			userWager3v3: 0,
+			newWager: 0,
+			userWager3v3Tanks: [null, null, null],
+			userWager1v1Tank: null,
+			newWager3v3Tanks: [],
+			newWager1v1Tank: getTank(blankTank),

```

Comment:
usually I would prefer if you could make this nullable rather than having a sentinel value for it

Reply:
I will make an issue for it, since we have it in other places. I'm going to move onto the battle arena implementation and testing for now.